### PR TITLE
graphql 3.0 and graphene 3.0 final rebase

### PR DESCRIPTION
**Pull request by ganwell**
I rebased almost all change to their original author: form #905 and #774.

### CODE REVIEW

**Comment by ganwell** [comment]
@jkimbo @ulgens is this PR too big to review?

File: setup.py [hunk 1]
```diff
@@ -53,9 +54,9 @@
     keywords="api graphql protocol rest relay graphene",
     packages=find_packages(exclude=["tests"]),
     install_requires=[
-        "graphene>=2.1.7,<3",
-        "graphql-core>=2.1.0,<3",
-        "Django>=1.11,!=2.0.*,!=2.1.*",
+        "graphene>=v3.0.0b1,<4",
+        "graphql-core>=v3.1.0,<4",
```
Comment (winged) [inline setup.py:58]: Hm. "v" prefix for versions?

**Comment by ganwell** [comment]
@jkimbo I reviewed the PR with my college @winged.

* I will fix setup.py

* And move the relevant commit message from all the v3-attempts into the main-commit

Then we really think, that can be merged into v3. Then I try v3 with caluma.

File: README.rst [hunk 2]
```diff
@@ -23,7 +23,7 @@ For installing graphene, just run this command in your shell
 
 .. code:: bash
 
-    pip install "graphene-django>=2.0"
+    pip install "graphene-django>=2.2"
```
Comment (jkimbo) [inline README.rst:26]:
```suggestion
    pip install "graphene-django>=3"
```

We'll be releasing this as graphene-django v3

**Comment by jkimbo** [comment]
@ganwell I've just merged the github actions migration into v3 so you'll have to rebase this change on top of that (sorry)

**Comment by ganwell** [comment]
@jkimbo this is hopfully the final rebase. I rebased all original commits from @ktosiek and @ulgens in to this PR. The diff to the version that was reviewed is trivial: https://git.io/Jfsh3

**Pull request merged**